packages/discovery-react-components/src/components/DocumentPreview/DocumentPreview.tsx

```diff
@@ -60,17 +65,61 @@ const DocumentPreview: FC<Props> = ({
     // setLoading(true);
   }, [doc]);
 
+  // fetch PDF (if necessary)
+  useEffect(() => {
+    async function fetchFile() {
+      const hasFile = await documentProvider!.provides(document!);
+      if (hasFile) {
+        setProvidedFile(await documentProvider?.get(document!));
+      }
+    }
+
+    // `file` takes precedence over a file provided by `documentProvider`
+    if (file) {
+      setProvidedFile(file);
+    } else if (document && documentProvider) {
+      fetchFile();
+    } else {
+      setProvidedFile(undefined);
+    }
+  }, [document, documentProvider, file]);
+
+  // TODO Seems like these 2 useEffects should just be part of PdfViewer
+  const [textMappings, setTextMappings] = useState<TextMappings | null>(null);
+  useEffect(() => {
+    const mappings = getTextMappings(doc);
+    if (mappings) {
+      setTextMappings(mappings);
+    }
+  }, [doc]);
+
+  // Pull total page count from either the PDF file or the structural
+  // data list
+  const [pageCount, setPageCount] = useState(0);
+  const [pdfPageCount, setPdfPageCount] = useState(pageCount);
+  useEffect(() => {
+    if (providedFile && pdfPageCount > 0) {
+      setPageCount(pdfPageCount);
+    } else if (textMappings) {
+      const last = textMappings.text_mappings.length - 1;
+      setPageCount(textMappings?.text_mappings[last].page.page_number ?? 1);
+    }
+  }, [textMappings, providedFile, pdfPageCount]);
+
   const base = `${settings.prefix}--document-preview`;
 
   return (
     <div className={`${base}`}>
-      {(doc || file) && !didCatch ? (
+      {(doc || providedFile) && !didCatch ? (
         <>
+          {/* TODO This is only used when showing PDFs (currently). Should be moved into `PdfViewer` or
+                    elsewhere that makes more sense.
+          */}
```

Comment:
@fukudasjp I think you were the one that pointed out the grey bar above all the files. This is it, the `PreviewToolbar`. I'm not sure how it fits in with your new redesign, but we should handle this better so it's only shown for PDFs, if that's only where it's needed.

Reply:
@jhpedemonte thanks for pointing this. I think about this toolbar and now I think that we can keep this toolbar code here. Because it's likely that the toolbar will have common action such as Search Text in document sooner or later. Instead, I will make the toolbar hidden when unnecessary.